crates/rome_js_analyze/src/analyzers/nursery/use_collapsed_else_if.rs

```diff
@@ -0,0 +1,143 @@
+use crate::JsRuleAction;
+use rome_analyze::{context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_syntax::{AnyJsStatement, JsBlockStatement, JsElseClause, JsIfStatement};
+use rome_rowan::{AstNode, AstNodeList, BatchMutationExt};
+
+declare_rule! {
+    /// Enforce using `else if` instead of `else { if ... }`.
+    ///
+    /// If an `if` statement is the only statement in the `else` block, it is often clearer to use an `else if` form.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lonely-if
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     } else {
+    ///         // ...
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else if (anotherCondition) {
+    ///     // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else if (anotherCondition) {
+    ///     // ...
+    /// } else {
+    ///     // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     }
+    ///     doSomething();
+    /// }
+    /// ```
+    ///
+    pub(crate) UseCollapsedElseIf {
+        version: "next",
+        name: "useCollapsedElseIf",
+        recommended: false,
+    }
+}
+
+pub(crate) struct RuleState {
+    block_statement: JsBlockStatement,
+    if_statement: JsIfStatement,
+    provide_fix: bool,
+}
+
+impl Rule for UseCollapsedElseIf {
+    type Query = Ast<JsElseClause>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let else_clause = ctx.query();
+        let alternate = else_clause.alternate().ok()?;
+        let block = alternate.as_js_block_statement()?;
+        let block_statements = block.statements();
+        if block_statements.len() != 1 {
+            return None;
+        }
+        if let AnyJsStatement::JsIfStatement(if_statement) = block_statements.first()? {
+            let has_comments = if_statement.syntax().has_leading_comments()
+                || block
+                    .r_curly_token()
+                    .is_ok_and(|token| token.has_leading_comments());
```

Comment:
You should also check if there are comments after the if statements and before the `l_curly` token.  

By the way, I suggest moving these checks in `action` where it is relevant.

Reply:
Just to be sure: You can replace it with `.ok()?` because it is ensured that `r_curly_token` is always `Ok`, right? Otherwise the function would return with `None` although 'has_comments' should only be `false`.

Ok, I thought the leading/trailing comments would overlap. Checked the playground, you're right!

Reply:
Missing `l_curly_token` and `r_curly_token` are errors. Usually lint rules aborts as soon as they encounter an error in the AST. Thus, you can just use `.ok()?`.